src/sentry/migrations/0024_auto_20191230_2052.py

```diff
@@ -79,7 +79,7 @@ def _attach_related(_events):
 
         try:
             group = event.group
-        except Group.DoesNotExist:
+        except (Group.DoesNotExist, NewGroup.DoesNotExist):
```

Comment:
Doesn't that actually mean that `e.group_id is None`?

Reply:
Yeah, I think it does. Might be able to just check on that before setting `group`.

Reply:
Are you sure this is the only case? What if they have an invalid group id set, which is entirely possible?

Reply:
This migration is a bit tricky in general since it uses so many external functions.

I think what I'd do here instead would be `group = Group.objects.get(id=event.group_id)` and not import `NewGroup` at all. The logic in `Event.group` really just handles caching which isn't going to be a thing here. We could also check that `group_id is not None` before making the query.

You'll also need to use `group` down below on `insert`, rather than `event.group`.

Reply:
@wedamija updated based on your suggestions. Rereview? <3